src/main/java/de/retest/surili/commons/core/TestCase.java

```
@@ -10,6 +10,11 @@
  */
 public interface TestCase extends Serializable {
 
+	/**
+	 * Default if, e.g., a particular test case name is not helpful.
+	 */
+	public static final String DEFAULT_TEST_CASE_NAME = "TestCase";
```

Is this really necessary within this interface, since you will only use this in the `TestCaseImpl`? A default method may be helpful, however it hides the fact that you should provide a unique name.

I think if you specify this within the interface, it gives the impression, that the test name is not necessary, thus might as well be `Optional<String> getName()`. What does your API define?

---

I first had it in the impl classes (which is why I forgot to remove `public`), but then switched to the interfaces. I actually like the idea of the name be an `Optional`, I think I'm gonna it that way!

---

Sure, It just depends on your API design.